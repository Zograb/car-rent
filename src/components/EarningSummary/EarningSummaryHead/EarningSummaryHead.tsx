import tw from 'twin.macro';


// Components
import { DatePicker } from '@/components/common/DatePicker';

export interface EarningSummaryHeadProps {
  startDate: Date;
  endDate: Date;
  minDate: Date;
  setStartDate: (date: Date) => void;
  setEndDate: (date: Date) => void;
}

const EarningSummaryHead = ({
  startDate,
  endDate,
  minDate,
  setStartDate,
  setEndDate,
}: EarningSummaryHeadProps) => {
  return (
    <div
      css={[
        tw`mb-6 text-sm`,
        tw`block justify-between items-center md:flex`,
      ]}
      data-testid="earning-summary-head"
    >
      <div tw="block justify-start items-center md:flex">
        <div tw="font-semibold mr-8 text-base">Earning Summary</div>
        <div tw="block justify-start items-center my-2 sm:flex md:my-0">
          <div tw="flex justify-start items-center">
            <div tw="mr-2 w-10 sm:w-auto">from:</div>
            <DatePicker
              css={[
                tw`border !border-primary/60 border-solid`,
                tw`w-28 p-0 h-auto text-center rounded`,
              ]}
              minDate={minDate}
              defaultDate={startDate}
              onChange={setStartDate}
            />
          </div>
          <div tw="flex justify-start items-center mt-3 sm:(ml-2 mt-0)">
            <div tw="mr-2 w-10 sm:w-auto">to:</div>
            <DatePicker
              css={[
                tw`w-28 p-0 h-auto text-center`,
                tw` border !border-primary/60 rounded border-solid`,
              ]}
              defaultDate={endDate}
              onChange={setEndDate}
            />
          </div>
        </div>
      </div>
      <div tw="flex justify-start items-center md:justify-end">
        <div tw="border-t-2 border-dashed border-gray3 w-4 h-px" />
        <div tw="ml-2 text-gray3">Same period last year</div>
      </div>
    </div>
  );
}

export default EarningSummaryHead;
