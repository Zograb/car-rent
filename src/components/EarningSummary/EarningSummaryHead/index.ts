export { type EarningSummaryHeadProps, default as EarningSummaryHead } from './EarningSummaryHead';
