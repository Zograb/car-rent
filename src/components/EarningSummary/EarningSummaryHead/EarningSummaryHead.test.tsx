import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

// Components
import EarningSummaryHead from './EarningSummaryHead';
import { format, subMonths } from 'date-fns';

describe('CustomTooltip', () => {
  // Temporarily workaround for bug in @testing-library/react when use user-event with `vi.useFakeTimers()`
  beforeAll(() => {
    const _jest = globalThis.jest;
    globalThis.jest = {
      ...globalThis.jest,
      advanceTimersByTime: jest.advanceTimersByTime,
    };
    return () => void (globalThis.jest = _jest);
  });

  beforeEach(() => {
    jest.useFakeTimers();
    jest.setSystemTime(new Date(2000, 12, 12));
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it('should match snapshot', () => {
    const { container } = render(
      <EarningSummaryHead
        startDate={new Date()}
        endDate={new Date()}
        minDate={new Date()}
        setStartDate={jest.fn()}
        setEndDate={jest.fn()}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('should render earning summary head', () => {
    const { getByTestId } = render(
      <EarningSummaryHead
        startDate={new Date()}
        endDate={new Date()}
        minDate={new Date()}
        setStartDate={jest.fn()}
        setEndDate={jest.fn()}
      />
    );

    const earningSummaryHead = getByTestId('earning-summary-head');
    expect(earningSummaryHead).toBeInTheDocument();
  });

  it('should render earning summary head with correct dates', () => {
    const endDate = new Date();
    const startDate = subMonths(new Date(), 4);
    const { getByDisplayValue } = render(
      <EarningSummaryHead
        startDate={startDate}
        endDate={endDate}
        minDate={startDate}
        setStartDate={jest.fn()}
        setEndDate={jest.fn()}
      />
    );

    const startDateInput = getByDisplayValue(format(startDate, 'MMM dd, yyy'));
    const endDateInput = getByDisplayValue(format(endDate, 'MMM dd, yyy'));
    expect(startDateInput).toBeInTheDocument();
    expect(endDateInput).toBeInTheDocument();
  });

  it('should call setStartDate and setEndDate function on inputs changes', async () => {
    const user = userEvent.setup({
      advanceTimers: jest.advanceTimersByTime,
    });
    const setStartDateFn = jest.fn();
    const setEndDateFn = jest.fn();
    const endDate = new Date();
    const startDate = subMonths(new Date(), 4);
    const { getByDisplayValue, findByText } = render(
      <EarningSummaryHead
        startDate={startDate}
        endDate={endDate}
        minDate={startDate}
        setStartDate={setStartDateFn}
        setEndDate={setEndDateFn}
      />
    );
    const startDateInput = getByDisplayValue(format(startDate, 'MMM dd, yyy'));
    const endDateInput = getByDisplayValue(format(endDate, 'MMM dd, yyy'));
    await user.click(startDateInput);
    await user.click(await findByText('14'));
    await user.click(endDateInput);
    await user.click(await findByText('16'));
    expect(setStartDateFn).toHaveBeenCalled();
    expect(setEndDateFn).toHaveBeenCalled();
  });
});
