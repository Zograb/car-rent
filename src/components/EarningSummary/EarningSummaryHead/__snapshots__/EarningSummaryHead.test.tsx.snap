// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTooltip should match snapshot 1`] = `
.c6 {
  position: relative;
  display: inline-block;
}

.c7 {
  position: absolute;
  left: 1rem;
  top: 50%;
  --tw-translate-y: -50%;
  transform: translate(var(--tw-translate-x), var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
  font-size: 1.5rem;
  line-height: 2rem;
  --tw-text-opacity: 1;
  color: rgb(101 101 117 / var(--tw-text-opacity));
}

.c8 {
  height: 3rem;
  border-radius: 0.5rem;
  border-width: 1px!important;
  border-style: solid!important;
  --tw-border-opacity: 1!important;
  border-color: rgb(222 222 222 / var(--tw-border-opacity))!important;
  --tw-text-opacity: 1;
  color: rgb(101 101 117 / var(--tw-text-opacity));
  outline: 2px solid transparent;
  outline-offset: 2px;
  padding-left: 1rem;
}

.c0 {
  margin-bottom: 1.5rem;
  font-size: 0.875rem;
  line-height: 1.25rem;
  display: block;
  align-items: center;
  justify-content: space-between;
}

.c1 {
  display: block;
  align-items: center;
  justify-content: flex-start;
}

.c2 {
  margin-right: 2rem;
  font-size: 1rem;
  line-height: 1.5rem;
  font-weight: 600;
}

.c3 {
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  display: block;
  align-items: center;
  justify-content: flex-start;
}

.c4 {
  display: flex;
  align-items: center;
  justify-content: flex-start;
}

.c5 {
  margin-right: 0.5rem;
  width: 2.5rem;
}

.c9 {
  border-width: 1px;
  border-style: solid;
  border-color: rgb(0 106 255 / 0.6)!important;
  height: auto;
  width: 7rem;
  border-radius: 0.25rem;
  padding: 0px;
  text-align: center;
}

.c10 {
  margin-top: 0.75rem;
  display: flex;
  align-items: center;
  justify-content: flex-start;
}

.c11 {
  margin-right: 0.5rem;
  width: 2.5rem;
}

.c12 {
  height: auto;
  width: 7rem;
  padding: 0px;
  text-align: center;
  border-radius: 0.25rem;
  border-width: 1px;
  border-style: solid;
  border-color: rgb(0 106 255 / 0.6)!important;
}

.c13 {
  display: flex;
  align-items: center;
  justify-content: flex-start;
}

.c14 {
  height: 1px;
  width: 1rem;
  border-top-width: 2px;
  border-style: dashed;
  --tw-border-opacity: 1;
  border-color: rgb(101 101 117 / var(--tw-border-opacity));
}

.c15 {
  margin-left: 0.5rem;
  --tw-text-opacity: 1;
  color: rgb(101 101 117 / var(--tw-text-opacity));
}

@media (min-width: 768px) {
  .c0 {
    display: flex;
  }
}

@media (min-width: 768px) {
  .c1 {
    display: flex;
  }
}

@media (min-width: 640px) {
  .c3 {
    display: flex;
  }
}

@media (min-width: 768px) {
  .c3 {
    margin-top: 0px;
    margin-bottom: 0px;
  }
}

@media (min-width: 640px) {
  .c5 {
    width: auto;
  }
}

@media (min-width: 640px) {
  .c10 {
    margin-left: 0.5rem;
    margin-top: 0px;
  }
}

@media (min-width: 640px) {
  .c11 {
    width: auto;
  }
}

@media (min-width: 768px) {
  .c13 {
    justify-content: flex-end;
  }
}

<div>
  <div
    class="c0"
    data-testid="earning-summary-head"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        Earning Summary
      </div>
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            from:
          </div>
          <div>
            <div
              class="react-datepicker-wrapper w-auto"
            >
              <div
                class="react-datepicker__input-container"
              >
                <div
                  class="c6"
                >
                  <div
                    class="c7"
                  />
                  <input
                    class="c8 c9"
                    value="Jan 12, 2001"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="c10"
        >
          <div
            class="c11"
          >
            to:
          </div>
          <div>
            <div
              class="react-datepicker-wrapper w-auto"
            >
              <div
                class="react-datepicker__input-container"
              >
                <div
                  class="c6"
                >
                  <div
                    class="c7"
                  />
                  <input
                    class="c8 c12"
                    value="Jan 12, 2001"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c13"
    >
      <div
        class="c14"
      />
      <div
        class="c15"
      >
        Same period last year
      </div>
    </div>
  </div>
</div>
`;
